test/functional/pdo_sqlsrv/pdo_ae_bindColumn_pdoparam_datetime.phpt

```diff
@@ -0,0 +1,76 @@
+--TEST--
+Test for retrieving encrypted data from datetime types columns using PDO::bindColumn
+--DESCRIPTION--
+Test conversion from datetime types column to output of PDO::PARAM types
+With or without AE, conversion works if:
+1. From any datetime type column to PDO::PARAM_STR
+2. From any datetime type column to PDO::PARAM_LOB
+--SKIPIF--
+<?php require('skipif_mid-refactor.inc'); ?>
+--FILE--
+<?php
+require_once("MsCommon_mid-refactor.inc");
+require_once("AEData.inc");
+
+$dataTypes = array("date", "datetime", "smalldatetime");
+
+try {
+    $conn = connect("", array(), PDO::ERRMODE_SILENT);
+    foreach ($dataTypes as $dataType) {
+        echo "\nTesting $dataType:\n";
+
+        // create and populate table containing date, datetime or smalldatetime columns
+        $tbname = "test_" . $dataType;
+        $colMetaArr = array(new ColumnMeta($dataType, "c_det"), new ColumnMeta($dataType, "c_rand", null, "randomized"));
+        createTable($conn, $tbname, $colMetaArr);
+        $inputValues = array_slice(${explode("(", $dataType)[0] . "_params"}, 1, 2);
+        insertRow($conn, $tbname, array("c_det" => $inputValues[0], "c_rand" => $inputValues[1]));
+        
+        // fetch by specifying PDO::PARAM_ types with PDO::bindColumn
+        $query = "SELECT c_det, c_rand FROM $tbname";
+        foreach ($pdoParamTypes as $pdoParamType) {
+            $det = "";
+            $rand = "";
+            $stmt = $conn->prepare($query);
+            $stmt->execute();
+            $stmt->bindColumn('c_det', $det, constant($pdoParamType));
+            $stmt->bindColumn('c_rand', $rand, constant($pdoParamType));
+            $row = $stmt->fetch(PDO::FETCH_BOUND);
+            
+            // check the case when fetching as PDO::PARAM_BOOL, PDO::PARAM_NULL or PDO::PARAM_INT
+            // with or without AE; should not work
+            if ($pdoParamType == "PDO::PARAM_BOOL" || $pdoParamType == "PDO::PARAM_NULL" || $pdoParamType == "PDO::PARAM_INT") {
```

Comment:
because for datetime, some input does not contain any decimal places. then value retrieved however, has 3 decimal places which are all 0 if the input did not have decimal places.